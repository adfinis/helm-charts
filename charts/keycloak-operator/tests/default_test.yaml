suite: default Keycloak Operator tests
templates:
  - operator/clusterrole.yaml
  - operator/deployment.yaml
  - operator/rolebinding.yaml
  - operator/roles.yaml
  - operator/serviceaccount.yaml
  - operator/service.yaml
tests:
  - it: should match snapshot
    values:
      - ../ci/default-values.yaml
    asserts:
      - matchSnapshot: {}
