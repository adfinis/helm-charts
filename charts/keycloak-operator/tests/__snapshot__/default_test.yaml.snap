should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloakcontroller-cluster-role
    rules:
      - apiGroups:
          - k8s.keycloak.org
        resources:
          - keycloaks
          - keycloaks/status
          - keycloaks/finalizers
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - patch
          - update
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloakrealmimportcontroller-cluster-role
    rules:
      - apiGroups:
          - k8s.keycloak.org
        resources:
          - keycloakrealmimports
          - keycloakrealmimports/status
          - keycloakrealmimports/finalizers
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - patch
          - update
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: RELEASE-NAME-keycloak-operator-operator
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: keycloak-operator
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: keycloak-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/component: keycloak-operator
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: keycloak-operator
        spec:
          containers:
            - env:
                - name: KUBERNETES_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: OPERATOR_KEYCLOAK_IMAGE
                  value: quay.io/keycloak/keycloak:23.0.2
              image: quay.io/keycloak/keycloak-operator:23.0.2
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                httpGet:
                  path: /q/health/live
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 30
                successThreshold: 1
                timeoutSeconds: 10
              name: keycloak-operator-operator
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /q/health/ready
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 30
                successThreshold: 1
                timeoutSeconds: 10
              resources: {}
              securityContext: {}
          securityContext: {}
          serviceAccountName: RELEASE-NAME-keycloak-operator
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloak-operator-role-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: keycloak-operator-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-keycloak-operator
  5: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloak-operator-view
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: view
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-keycloak-operator
  6: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloakcontroller-role-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: keycloakcontroller-cluster-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-keycloak-operator
  7: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloakrealmimportcontroller-role-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: keycloakrealmimportcontroller-cluster-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-keycloak-operator
  8: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: keycloak-operator-role
    rules:
      - apiGroups:
          - apps
          - extensions
        resources:
          - statefulsets
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - patch
          - update
      - apiGroups:
          - ""
        resources:
          - secrets
          - services
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - patch
          - update
      - apiGroups:
          - batch
        resources:
          - jobs
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - patch
          - update
      - apiGroups:
          - networking.k8s.io
        resources:
          - ingresses
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - patch
          - update
  9: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: RELEASE-NAME-keycloak-operator-operator
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: keycloak-operator
      type: ClusterIP
  10: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: keycloak-operator
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 23.0.2
        helm.sh/chart: keycloak-operator-1.0.2
      name: RELEASE-NAME-keycloak-operator
